### PR TITLE
FIX broken tests: how's the test ressource supposed to be found if we cut the path ?

### LINKED ISSUE #17: Broken tests in psc-extract
Some tests in psc-extract are broken due to apparent path computation failures.

[ERROR] Tests run: 7, Failures: 0, Errors: 4, Skipped: 1, Time elapsed: 2.643 s <<< FAILURE! - in fr.ans.psc.pscextract.ExtractionControllerTests
[ERROR] singlePageExtractionAndResultConformityTest  Time elapsed: 0.131 s  <<< ERROR!
java.nio.file.NoSuchFileException: home/ericdegenetais/projects/ANS/psc-ecosystem/psc-extract/target/test-classes/wiremock/__files/multiple-work-situations-result.txt
	at fr.ans.psc.pscextract.ExtractionControllerTests.singlePageExtractionAndResultConformityTest(ExtractionControllerTests.java:104)

[ERROR] fullyEmptyPsExtractionTest  Time elapsed: 0.106 s  <<< ERROR!
java.nio.file.NoSuchFileException: home/ericdegenetais/projects/ANS/psc-ecosystem/psc-extract/target/test-classes/wiremock/__files/very-empty-ps-result.txt
	at fr.ans.psc.pscextract.ExtractionControllerTests.fullyEmptyPsExtractionTest(ExtractionControllerTests.java:190)

[ERROR] multiplePagesExtractionAndResultConformityTest  Time elapsed: 0.305 s  <<< ERROR!
java.nio.file.NoSuchFileException: home/ericdegenetais/projects/ANS/psc-ecosystem/psc-extract/target/test-classes/wiremock/__files/multiple-pages-result.txt
	at fr.ans.psc.pscextract.ExtractionControllerTests.multiplePagesExtractionAndResultConformityTest(ExtractionControllerTests.java:127)

[ERROR] generateExtractTest  Time elapsed: 0.105 s  <<< ERROR!
java.nio.file.NoSuchFileException: home/ericdegenetais/projects/ANS/psc-ecosystem/psc-extract/target/test-classes/wiremock/__files/multiple-pages-result.txt
	at fr.ans.psc.pscextract.ExtractionControllerTests.generateExtractTest(ExtractionControllerTests.java:213)

### PR DESCRIPTION
**Pull request by edegenetais-nx**
Closes #17

### CODE REVIEW

File: src/test/java/fr/ans/psc/pscextract/ExtractionControllerTests.java [hunk 1]
```diff
@@ -99,49 +99,34 @@ private void Clean() {
...
-
     await().until(controllerIsReady(controller));
 
     ResponseEntity<FileSystemResource> response = controller.getFile();
 
-    byte[] responseBytes = getBytesFromResponse(response);
-
-    String expected = new String(expectedResponseBytes, StandardCharsets.UTF_8);
-    String actual = new String(responseBytes, StandardCharsets.UTF_8);
-
```
Comment (bou3108) [inline src/test/java/fr/ans/psc/pscextract/ExtractionControllerTests.java:119]: Two problems here : 
- the resource.getPath() method leaves an invalid character (<:>) at index 2 while running on Windows environment. Ref: https://stackoverflow.com/questions/43972777/java-nio-file-invalidpathexception-illegal-char-at-index-2
That's why I used to call substring(1), which isn't a clean solution => I changed it to a File based approach (see getContentAsString method below)
- EOL errors in expected results files : add a getNormalizedEOL() method to neutralize Windows EOL conversions.

File: src/test/resources/wiremock/__files/empty-ps-result.txt [hunk 2]
```diff
@@ -1,7 +1 @@
...
-||||''|||||||||||||||||||||||||||||||||||||||||||||||
-||||''|||||||||||||||||||||||||||||||||||||||||||||||
-||||''|||||||||||||||||||||||||||||||||||||||||||||||
-||||''|||||||||||||||||||||||||||||||||||||||||||||||
-||||''|||||||||||||||||||||||||||||||||||||||||||||||
-||||''|||||||||||||||||||||||||||||||||||||||||||||||
```
Comment (bou3108) [inline src/test/resources/wiremock/__files/empty-ps-result.txt:7]: The previous file was very odd. The service has no reason to generate an extract with 7 lines and separators from an empty response !

File: .github/workflows/run-tests.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: run tests on pull requests
+
+on:
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
```
Comment (bou3108) [inline .github/workflows/run-tests.yml]: Seems useful to run tests on each push in the pull request, and not only at the merging in main step ? What do you think ?

**Comment by edegenetais-nx** [comment]
Par contre il a l'air chafouin. Je vais, pour commencer, refaire un test linux de mon côté.

**Review by bou3108 — APPROVED** [review]
Fixes Linux / Windows inconsistencies.
Runs on my local M$ as well as in Ubuntu Github runner

**Comment by edegenetais-nx** [comment]
> Fixes Linux / Windows inconsistencies. Runs on my local M$ as well as in Ubuntu Github runner

Works on my linux workstation too. I bet we're ready to merge that one !

**Pull request merged**